src/Hosting/Hosting/src/Internal/HostingMetrics.cs

```diff
@@ -0,0 +1,42 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.Metrics;
+using Microsoft.Extensions.Metrics;
+
+namespace Microsoft.AspNetCore.Hosting;
+
+internal class HostingMetrics : IDisposable
+{
+    private readonly Meter _meter;
+    private readonly ObservableCounter<long> _totalRequestsCounter;
+    private readonly ObservableUpDownCounter<long> _currentRequestsCounter;
+    private readonly ObservableCounter<long> _failedRequestsCounter;
+
+    public HostingMetrics(IMetricsFactory metricsFactory)
+    {
+        _meter = metricsFactory.CreateMeter("Microsoft.AspNetCore.Hosting");
+
+        _totalRequestsCounter = _meter.CreateObservableCounter<long>(
+            "total-requests",
```

Comment:
Meter name is the same as the event source name. Metrics counter name is the same as the event source counter name. I recall that only one would show up in tooling like dotnet-counters. Is that right?

What happens if a name is shared, but the metrics counter adds tags?

Reply:
If the name is shared, the Metrics one will win (so says @noahfalk)

Reply:
Suggested dimensions: Http Version, Http Method, Route Path (eg `/products/{productid}/details`)

Reply:
There is also HTTP status code.

I was warned about creating an explosion of tags. I don't think we can log all those dimensions together in one counter.